aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -67,10 +67,13 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
   delete [] idx;
 }
 
-#if !defined(TH_REAL_IS_BOOL) /* non bool only part */
-
-void THTensor_(maskedFill)(THTensor *tensor, THByteTensor *mask, scalar_t value)
+void THTensor_(maskedFill)(THTensor *tensor, THBoolTensor *mask, scalar_t value)
 {
+  // TODO: remove ScalarType::Byte path after the release as deprecated.
+  if (mask->dtype() == at::ScalarType::Byte) {
```

Comment:
this is pretty confusing -- why does this take a THBoolTensor but that can be a byte?  Instead, why not just write overloads (or give them different names, whatever works better), for Bool and Byte versions?  They can call into the same implementation after the checking is done.

Reply:
Initially i wanted to add an overload version which would be taking THBoolTensor mask, but no matter which mask i would pass, bool or byte, it will still go through the THByteTensor version and fail due to storage type errors. Long story short, @ezyang, helped me figure out that there is a dispatcher which decides which overload to use and in order to make it work with overloads, we have to add a new dispatcher. This solution seems a bit too complex for a given problem assuming that Byte masks will be killed soon. 

So thats why i've added checked_bool_tensor_unwrap which has to work for Bool and Byte types.